src/ptvsd/messaging.py

```diff
@@ -100,7 +100,10 @@ def read_json(self):
             raise IOError('Content-Length is missing or invalid')
 
         try:
-            body = self._reader.read(length)
+            body = b''
+            while length:
+                body += self._reader.read(length)
+                length -= len(body)
```

Comment:
length calculation is wrong here. May be `while length > len(body):` should do it?